File: java/Taskie-retrofit-fragments/app/src/main/java/ada/osc/taskie/model/TaskGenerator.java
package ada.osc.taskie.model;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import ada.osc.taskie.model.Task;
import ada.osc.taskie.model.TaskPriority;

public class TaskGenerator {

	private static final Random generator = new Random();

	private static String[] titles = {
			"Osc", "Potato", "Homework", "Shopping", "Gaming", "Reading",
			"Android", "Studying", "Question", "Ideas", "Party", "Nothing",
	};

	private static String[] descriptions = {
			"Do it.", "Play it", "Drink it", "Answer it", "Shut it down",
			"Try it.", "Don't do it.", "Ignore it.",
	};

	public static List<Task> generate(int taskCount) {
		List<Task> tasks = new ArrayList<Task>();
		for (int i=0; i<taskCount; i++){
			String title = titles[generator.nextInt(titles.length)];
			String description = descriptions[generator.nextInt(descriptions.length)];

			int prioritySelector = generator.nextInt(TaskPriority.values().length);
			TaskPriority priority = TaskPriority.values()[prioritySelector];

			tasks.add(new Task(title, description, priority));
		}
		return tasks;
	}
}

File: kotlin/Taskie/app/src/main/java/ada/osc/taskie/model/TaskGenerator.kt
package ada.osc.taskie.model

import java.util.*

object TaskGenerator {

    private val generator = Random()

    private val titles = arrayOf("Osc", "Potato", "Homework", "Shopping", "Gaming", "Reading", "Android", "Studying", "Question", "Ideas", "Party", "Nothing")
    private val descriptions = arrayOf("Do it.", "Play it", "Drink it", "Answer it", "Shut it down", "Try it.", "Don't do it.", "Ignore it.")

    fun generate(taskCount: Int): List<Task> = (0..taskCount).map { counter ->
        val title = titles[generator.nextInt(titles.size)]
        val description = descriptions[generator.nextInt(descriptions.size)]

        val prioritySelector = generator.nextInt(TaskPriority.values().size)
        val priority = TaskPriority.values()[prioritySelector]

        Task(id = counter, title = title, description = description, priority = priority)
    }
}

